### ISSUE TITLE
0.56.x (latest: 0.56.0)

### ISSUE THREAD

**Issue by grabbou** [react-native-community/react-native-releases#15]
0.56.0

**Comment by grabbou** [comment]
An update: I have lost my laptop temporarily and setting up a new dev environment today. I will try to cut new release today if no `master` errors happen in the meantime. Otherwise, it will happen tomorrow.

Would be great if people that are waiting or somewhat blocked by this could "+1" this message so that I have a better overview of the consumption rate.

**Comment by grabbou** [comment]
Unfortunately, the `master` is still broken with some issues related to Babel (who knows what happens when we fix them).

Pinged the core team for the first time after a while to see where we are.

**Comment by hramos** [comment]
We're working hard to resolve the remaining test failures in `master`. Here's a summary of where we're at.

## `test_android` is failing

Our Android tests break often, for various reasons. It has a large surface area, and changes related to BUCK, Android, or JavaScript can all introduce a failure. Once these tests go red, it's easy to lose track of new failures and getting back to green can take a while. This is where we're at.

After fixing some BUCK issues and reverting to Android SDK 23, the latest blocker is a failure to build the JavaScript bundle necessary to run our unit tests on Android. This should be resolved once `jest-haste-map` is updated to support SHA-1 generation for haste modules in `node_modules` (https://github.com/facebook/react-native/pull/18995#pullrequestreview-118398614).

## `test_javascript` is failing

The latest failure can be attributed to some failing snapshot tests, following the release of `jest@23.0.0-charlie.2`. Undefined props are no longer printed out to snapshots. This should be resolved by updating our snapshots, but Facebook is currently using `jest@23.0.0-charlie.1` in the internal monorepo that contains React Native. Once that codebase moves to `jest@23.0.0`, we should be able to unblock ourselves here.

There's still one more failing test that needs to be addressed before we can get back to green:

```
Summary of all failing tests
 FAIL  Libraries/Animated/src/__tests__/Animated-test.js
  ● Animated tests › Animated › send toValue when an underdamped spring stops

    Ran 100000 timers, and there are still more! Assuming we've hit an infinite recursion and bailing out...

      140 |       anim.addListener(listener);
      141 |       Animated.spring(anim, {toValue: 15}).start();
    > 142 |       jest.runAllTimers();
          |            ^
      143 |       const lastValue =
      144 |         listener.mock.calls[listener.mock.calls.length - 2][0].value;
      145 |       expect(lastValue).not.toBe(15);

      at FakeTimers.runAllTimers (node_modules/jest-util/build/fake_timers.js:194:13)
      at Object.<anonymous> (Libraries/Animated/src/__tests__/Animated-test.js:142:12)
```

If you're interested in seeing 0.56.x getting released, please consider how you may be able to help in resolving the issues outlined above.

**Comment by hramos** [comment]
`test_javascript` is now fixed. Once `test_android` and the Objective-C e2e tests are green, we should be good to go with 0.56. Contributions are welcome.

**Comment by hramos** [comment]
All tests green and ready to go.

**Comment by hramos** [comment]
[0.56.0-rc.0](https://github.com/facebook/react-native/tree/v0.56.0-rc) is now public.

**Comment by grabbou** [comment]
Thanks! I was going to do it now, but since you have done it already (big
thanks!), i will move to other stuff in the meantime.

**Comment by grabbou** [comment]
Latest version as of now is: 0.56.0-rc.2.

**Comment by grabbou** [comment]
**Status**: CI tests pending
**Date**: 21/06/2018
**Next update**: 21/06/2018
**Expected status**: RC published

**Comment by grabbou** [comment]
**Status**: RC published (0.56.0-rc.4)
**Date**: 21/06/2018

**Comment by grabbou** [comment]
Status: Stable published (0.56.0)
Date: 4/07/2018
